wiktionary_verb_scraper.py

```diff
@@ -0,0 +1,127 @@
+import argparse
+import csv
+import unicodedata
+from collections import defaultdict
+
+import requests
+from bs4 import BeautifulSoup
+
+
+def query_pages():
+    params = defaultdict(set, {
+        'format': 'json',
+        'action': 'query',
+        'list': 'categorymembers',
+        'cmtitle': 'Категория:Русские_глаголы',
+    })
+
+    while True:
+        resp = requests.get(
+            'https://ru.wiktionary.org/w/api.php',
+            params=params
+        ).json()
+
+        if 'error' in resp:
+            raise ValueError(resp['error'])
+        if 'warnings' in resp:
+            print(resp['warnings'])
+        if 'query' in resp:
+            yield resp['query']
+        if 'continue' not in resp:
+            break
+        params.update(resp['continue'])
+
+
+def get_page(page_id):
+    url = 'https://ru.m.wiktionary.org/wiki/'
+    return requests.get(
+        url=url,
+        params={
+            'curid': page_id
+        }
+    ).content
+
+
+def get_verb(doc):
+    soup = BeautifulSoup(doc, 'html.parser')
+    columns = soup.findAll('td', {
+        'align': 'left'
+    })
+
+    forms = []
+    for item in columns:
+        forms += map(
+            lambda x: sanitize(x).strip(),
+            filter(lambda x: isinstance(x, str), item.contents)
+        )
+
+    return {
+        'present_i': forms[0],
+        'present_thou': forms[4],
+        'present_it': forms[8],
+
+        'present_we': forms[13],
+        'present_you': forms[16],
+        'present_they': forms[19],
+
+        'past_male': forms[9],
+        'past_female': forms[10],
+        'past_neutral': forms[11],
+        'past_many': forms[14],
+        'imperative': forms[7]
+    }
+
+
+def sanitize(word):
+    return ''.join(filter(
+        # we can use set instead of list here, but meh...
```

Comment:
Well, I don't think this comment is significantly useful here. Could you remove it please?

Reply:
done